packages/react-test-renderer/src/__tests__/ReactTestRenderer-test.js

```diff
@@ -120,5 +120,64 @@ describe('ReactTestRenderer', () => {
       await waitForAll([]);
       expect(root.toJSON().children).toEqual(['dynamic']);
     });
+
+    it('failing test', async () => {
+      const SELECT_ALL_FILTERS_VALUE = '_selected_all_filters_';
+      function Parent() {
+        const [selectedFilters, setSelectedFilters] = React.useState(null);
+        const filterList = ['green', 'red', 'gray'];
+        return (
+          <Child
+            selectedFilters={selectedFilters}
+            setSelectedFilters={setSelectedFilters}
+            filterList={filterList}
+          />
+        );
+      }
+
+      function Child({selectedFilters, setSelectedFilters, filterList}) {
+        const selectedFiltersString = React.useMemo(() => {
+          if (selectedFilters == null) {
+            return '';
+          }
+          if (selectedFilters.length === 1) {
+            return selectedFilters[0];
+          }
+          return SELECT_ALL_FILTERS_VALUE;
+        }, [selectedFilters]);
+
+        const onFilterChange = React.useCallback(
+          filter => {
+            if (filter === SELECT_ALL_FILTERS_VALUE) {
+              setSelectedFilters(filterList);
+            } else {
+              setSelectedFilters([filter]);
+            }
+          },
+          [filterList, setSelectedFilters],
+        );
+
+        React.useEffect(() => {
+          if (filterList.length === 0 && selectedFilters != null) {
+            onFilterChange(null);
+          } else if (filterList.length > 0) {
+            if (
+              selectedFilters == null ||
+              (selectedFilters.length === 1 &&
+                !filterList.includes(selectedFilters[0])) ||
+              (selectedFilters.length > 1 &&
+                JSON.stringify(selectedFilters) !== JSON.stringify(filterList))
+            ) {
+              onFilterChange(SELECT_ALL_FILTERS_VALUE);
+            }
+          }
+        }, [filterList, selectedFilters, onFilterChange]);
+
+        return <div>Selected Filters: {selectedFiltersString}</div>;
+      }
+
+      const root = ReactTestRenderer.create(<Parent />);
+      expect(root.toJSON()).toEqual('_selected_all_filters_');
```

Comment:
wouldn't the expected output be a `<div>Selected filters: ...</div>` or similar?

Reply:
Simplified it, it should basically hit the `selectedFilters == null` condition and set it to the constant string